### PR TITLE
executor: make the format of user and host fields ... (#18162)

### PR DESCRIPTION
**Pull request by ti-srebot**
cherry-pick #18162 to release-4.0

---

### What problem does this PR solve?

Issue Number: related to #17846 

Problem Summary: make the format of user and host fields in slow logs be compatible with MySQL

### What is changed and how it works?
The format of TiDB now is: `# User: {user_name}@{host_address}`
For example: `# User: root@127.0.0.1`

After this PR, it will be: `# User@Host: {user_name}[{user_name}] @ {host_name} [{host_address}]`, which is compatible with MySQL.
For example: `# User@Host: root[root] @ localhost [127.0.0.1]`

### Related changes

- PR to update `pingcap/docs`/`pingcap/docs-cn`:

### Check List 

Tests 

- Unit test

### Release note 

- executor: make the format of user and host fields in slow logs be compatible with MySQL

### CODE REVIEW

File: executor/slow_query_test.go [hunk 1]
```diff
@@ -76,7 +77,11 @@ select * from t;`
 		}
 		recordString += str
 	}
+<<<<<<< HEAD
```
Comment (SunRunAway) [inline executor/slow_query_test.go:80]: Please resolve this conflict.

**Review by zz-jason — COMMENTED** [review]
@qw4990 please resolve the conflicts and fix CI.

**Comment by qw4990** [comment]
PTAL @zz-jason @SunRunAway

**Review by zz-jason — APPROVED** [review]
LGTM

**Pull request merged**